tests/reg_tests/test_DVGeometryMulti.py

```diff
@@ -0,0 +1,222 @@
+            nTwist = nRefAxPts - 1
+
+            # Set up a twist variable
+            def twist(val, geo):
+                for i in range(1, nRefAxPts):
+                    geo.rot_z["box"].coef[i] = val[i - 1]
+
+            DVGeoDict[comp].addGlobalDV(dvName=f"{comp}_twist", value=[0] * nTwist, func=twist)
+
+        # Define a test point set
+        pts = np.array(
+            [
+                # Points on box1 away from intersection
+                [0.0, 0.0, 0.0],
+                [0.5, 0.1, -0.5],
+                # Points on box2 away from intersection
+                [0.5, 0.0, 2.0],
+                [0.375, 0.125, 2.0],
+                # Point on box3
+                [2.5, 0.5, 0.0],
+                # Point on curve part_15_1d
+                [0.3, 0.0, 0.5],
+                # Point on curve part_35_1d
+                [0.75, -0.25, 0.6],
+                # Point on curve part_37_1d
+                [0.25, -0.25, 0.6],
+                # Point on curve part_39_1d
+                [0.25, 0.25, 0.6],
+                # Point on intersection curve
+                [0.25, 0.1, 0.5],
+                # Point on tracked surface part_14, box1
+                [0.5, -0.3, 0.5],
+                # Point on tracked surface part_39, box2
+                [0.25, 0.12, 0.51],
+                # Point on excluded surface part_15, box1
+                [0.5, 0.3, 0.5],
+                # Point on excluded surface part_40, box2
+                [0.375, 0.25, 0.6],
+                # Other points near the intersection
+                [0.25, 0.251, 0.5],
+                [0.5, 0.251, 0.5],
+                [0.51, 0.25, 0.4],
+                [0.75, 0.25, 0.6],
+                [0.5, 0.25, 0.6],
+                [0.25, 0.5, 0.6],
+                [0.5, -0.25, 0.6],
+                [0.25, -0.5, 0.6],
+            ]
+        )
+
+        # Add the point set
+        ptSetName = "test_set"
+        comm = MPI.COMM_WORLD
+        DVGeo.addPointSet(pts, ptSetName, comm=comm, applyIC=True)
+
+        # Apply twist to the two intersecting boxes
+        dvDict = DVGeo.getValues()
+        dvDict["box1_twist"] = 2
+        dvDict["box2_twist"] = 2
+        DVGeo.setDesignVars(dvDict)
+
+        # Update the point set
+        ptsUpdated = DVGeo.update(ptSetName)
+
+        # Regression test the updated points
+        refFile = os.path.join(baseDir, "ref/test_DVGeometryMulti.ref")
+        with BaseRegTest(refFile, train=train) as handler:
+            handler.par_add_val("ptsUpdated", ptsUpdated, tol=1e-14)
+
+        # Now we will test the derivatives
+
+        # Build a dIdpt array
+        # We will have nNodes*3 many functions of interest
+        nNodes = pts.shape[0]
+        dIdpt = np.zeros((nNodes * 3, nNodes, 3))
+
+        # Set the seeds to one in the following fashion
+        # The first function of interest gets the first coordinate of the first point
```

Comment:
It took me a bit to figure out what is going on here, we are basically getting "individual" derivative outputs for every node coordinate, right? I am not sure if it is worth adding some clarification, especially on a test script though

Reply:
Yes, that's correct. I will add a comment to clarify this.